modules/SerialCommHub/tiny_modbus_rtu.cpp

```diff
@@ -438,7 +487,17 @@ std::vector<uint16_t> TinyModbusRTU::txrx_impl(uint8_t device_address, FunctionC
 
         // write to serial port
         rxtx_gpio.set(false);
-        write(fd, req.data(), req.size());
+
+        uint8_t* buffer = req.data();
+        ssize_t written = 0;
+
+        while (written < req.size()) {
+            ssize_t c = write(fd, &buffer[written], req.size() - written);
+            if (c == -1)
+                throw std::system_error(errno, std::generic_category(), "Could not send Modbus request");
```

Comment:
That's the system_error that was mentioned earlier that should be avoided

Reply:
Yes, but now it's being caught. I that not okay?

Reply:
ah somehow I missed that :sweat_smile: all good